sktime/benchmarking/benchmarks.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+"""Benchmarking interface for use with sktime objects.
+
+Wraps kotsu benchmarking interface.
+"""
+from typing import Callable, Optional, Type, Union
+
+import kotsu
+
+from sktime.base import BaseEstimator
+
+
+class BaseBenchmark:
+    """Base class for benchmarks.
+
+    A benchmark consists of a set of tasks and a set of estimators.
+    """
+
+    def __init__(self):
+        self.estimators = kotsu.registration.ModelRegistry()
+        self.validations = kotsu.registration.ValidationRegistry()
+
+    def add_estimator(
+        self,
+        estimator_entrypoint: Type[BaseEstimator],
+        estimator_kwargs: Optional[dict] = None,
+        estimator_id: Optional[str] = None,
+    ):
+        """Register an estimator to the benchmark."""
+        estimator_id = estimator_id or f"{estimator_entrypoint.__name__}-v1"
+        self.estimators.register(
+            id=estimator_id, entry_point=estimator_entrypoint, kwargs=estimator_kwargs
+        )
+
+    def _add_task(
```

Comment:
Having considered this, I can't figure out how this pattern can be usefully applied here, at least not in the same way it is usefully applied to estimators. 

For forecasters:
```
class BaseForecaster:
    ...
    def fit(self, y, X=None, fh=None):
        .... # do some data validation etc.
        self._fit(y=y_inner, X=X_inner, fh=fh)
        ...

...

class _ProphetAdapter(BaseForecaster):
    ...
    def _fit(self, y, X=None, fh=None):
        ...
```
is useful because `fit` and all subclass `_fit`s have the same signature. 

For benchmarkers, the signatures are not the same, as the task parameters are specific to the subclass (in the same way the init parameters are specific to the forecaster subclasses). The only way I could see this working is for `BaseBenchmark`'s `add_task` to take something like `*args, **kwargs`, but then with no assurances on what the input to the function is there can't be any reliable pre-processing/checks that is dependent on the inputs before `_add_task` is called as happens for `fit`. 

Could you advise if I'm missing something here?

Reply:
hm, I suppose you are right.

Originally, I thought, all the tasks would have comparable signature, but I guess in the current design they do not.